CharlieBackend.Api.UnitTes/DashboardServiceTest.cs

```diff
@@ -1,22 +1,97 @@
 using CharlieBackend.Core.DTO.Dashboard;
 using CharlieBackend.Core.Entities;
+using CharlieBackend.Core.Mapping;
 using CharlieBackend.Core.Models.ResultModel;
 using CharlieBackend.Data.Repositories.Impl.Interfaces;
 using Moq;
 using Xunit;
+using FluentAssertions;
 
 namespace CharlieBackend.Api.UnitTest
 {
     public class DashboardServiceTest : TestBase
     {
+        private readonly Mock<IDashboardRepository> _dashboardRepositoryMock;
+        private readonly IMapper _mapper;
+        private static long studentIdWithGroup = 7;
+        private static long studentIdWithoutGroup = 20;
+        private static long courseId = 1;
+        private static long courseIdWithoutGroup = 2;
+
+        public DashboardServiceTest()
+        {
+            _dashboardRepositoryMock = new Mock<IDashboardRepository>();
+            _mapper = GetMapper(new ModelMappingProfile());
+        }
+
+        private void InitializeForDashboardRepository()
+        {
+            _unitOfWorkMock.Setup(x => x.DashboardRepository).Returns(_dashboardRepositoryMock.Object);
+            _currentUserServiceMock = GetCurrentUserAsExistingStudent();
+        }
+
+        [Fact]
+        public async Task GetStudentsClassbook_ValidDataPassed_ShouldReturnCurrentStudentsMarks()
+        {
+            //Arrange
+            var studentclassbookRequestWithData = new StudentsRequestDto<ClassbookResultType>()
+            {
+                CourseId = 1,
+                StudentGroupId = 0,
+                StartDate = new DateTime(2000, 01, 01),
+                FinishDate = new DateTime(2030, 01, 01),
+                IncludeAnalytics = new ClassbookResultType[]
+                {
+                    ClassbookResultType.StudentPresence,
+                    ClassbookResultType.StudentMarks
+                }
+            };
+
+            var expectedStudentsMarks = new List<StudentMarkDto>
+            {
+                new StudentMarkDto()
+                    {
+                        CourseId = 2,
+                        StudentGroupId = 1,
+                        StudentId = 5,
+                        LessonId = 2,
+                        LessonDate = new DateTime(2015,4,12),
+                        StudentMark = 5
+                    }
+            };
+
+            _dashboardRepositoryMock.Setup(x => x.GetGroupsIdsByCourseIdAndPeriodAsync(
+                studentclassbookRequestWithData.CourseId.Value,
+                studentclassbookRequestWithData.StartDate,
+                studentclassbookRequestWithData.FinishDate)).ReturnsAsync(new List<long>() { 2 });
+
+            _dashboardRepositoryMock.Setup(x => x.GetStudentsMarksListByGroupIdsAndDate(
+                new List<long> { 2 }, studentclassbookRequestWithData.StartDate, studentclassbookRequestWithData.FinishDate))
+                .ReturnsAsync(expectedStudentsMarks);
+
+            InitializeForDashboardRepository();
+
+            var dashboardService = new DashboardService(
```

Comment:
Create a field _dashboardService and initialize it in the constructor. Then you will be able to configure _unitOfWorkMock and dashboardService will know about these changes as it is reference type passed.

Reply:
Solved like this https://github.com/ita-social-projects/WhatBackend/pull/409/commits/8f6a49149d50761f3377e73f420db31c754c8bce